services/src/projects/project.rs

```diff
@@ -253,10 +264,57 @@ pub struct UpdateProject {
     pub id: ProjectId,
     pub name: Option<String>,
     pub description: Option<String>,
-    pub layers: Option<Vec<Option<Layer>>>,
+    pub layers: Option<Vec<LayerUpdate>>,
     pub bounds: Option<STRectangle>,
 }
 
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+#[serde(untagged)]
+pub enum LayerUpdate {
```

Comment:
Need third variant: `Add` because it doesn't seem natural to Add a (first) layer to a project by updating a non-existing one.

Reply:
What is the advantage of that type? I guess we would only throw more errors without enhancing the functionality.

Reply:
it makes it easier to catch bugs in the frontend where it wants to update layers that do not exist. Also makes the API more intuitive. If you look at your test it says "add first layer" but creates an Update-object

Reply:
I could rename it to upsert, because what is the difference between adding and updating after all? We store a new layer object in that place. My problem with it is that it does not provide real added value. Both variants do the same, just one crashes on one occasion. Maybe we will introduce it if we allow partial updates on layers some time.